sklearn/utils/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,300 @@
+from __future__ import print_function
+
+from nose.tools import assert_true
+import inspect
+import warnings
+import importlib
+
+from pkgutil import walk_packages
+from inspect import getsource
+
+import sklearn
+from sklearn.utils.testing import SkipTest
+from sklearn.utils.testing import assert_raise_message
+
+_doc_special_members = ('__contains__', '__getitem__', '__iter__', '__len__',
+                        '__call__', '__add__', '__sub__', '__mul__', '__div__',
+                        '__neg__', '__hash__')
+
+public_modules = [
+    # the list of modules users need to access for all functionality
+    # 'sklearn',
+    # 'sklearn.cluster',
+    # 'sklearn.covariance',
+    # 'sklearn.cross_decomposition',
+    'sklearn.datasets',
+    # 'sklearn.decomposition',
+    # 'sklearn.ensemble',
+    # 'sklearn.feature_extraction',
+    # 'sklearn.feature_selection',
+    # 'sklearn.gaussian_process',
+    # 'sklearn.linear_model',
+    # 'sklearn.manifold',
+    # 'sklearn.metrics',
+    # 'sklearn.mixture',
+    # 'sklearn.model_selection',
+    # 'sklearn.neighbors',
+    # 'sklearn.neural_network',
+    # 'sklearn.preprocessing',
+    # 'sklearn.semi_supervised',
+    # 'sklearn.tree',
+    # 'sklearn.utils',
+]
+
+
+# helpers to get function arguments
+if hasattr(inspect, 'signature'):  # py35
+    def _get_args(function, varargs=False):
+        params = inspect.signature(function).parameters
+        args = [key for key, param in params.items()
+                if param.kind not in (param.VAR_POSITIONAL, param.VAR_KEYWORD)]
+        if varargs:
+            varargs = [param.name for param in params.values()
+                       if param.kind == param.VAR_POSITIONAL]
+            if len(varargs) == 0:
+                varargs = None
+            return args, varargs
+        else:
+            return args
+else:
+    def _get_args(function, varargs=False):
+        out = inspect.getargspec(function)  # args, varargs, keywords, defaults
+        if varargs:
+            return out[:2]
+        else:
+            return out[0]
+
+
+def get_name(func):
+    parts = []
+    module = inspect.getmodule(func)
+    if module:
+        parts.append(module.__name__)
+    if hasattr(func, 'im_class'):
+        parts.append(func.im_class.__name__)
+    parts.append(func.__name__)
+    return '.'.join(parts)
+
+
+# functions to ignore args / docstring of
+_docstring_ignores = [
+    'sklearn.utils.deprecation.load_mlcomp',
+]
+
+_tab_ignores = [
+]
+
+
+def check_parameters_match(func, doc=None):
```

Comment:
This file currently contains three things from what I can see: a function to check docstrings, tests for this function, and tests that run this function on sklearn. I feel they should live in two or three different files since they are conceptually very separate.